dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/queue/PeerTaskInstancePriorityQueue.java

```diff
@@ -61,6 +62,22 @@ public TaskInstance take() throws TaskPriorityQueueException {
         return queue.poll();
     }
 
+    /**
+     * poll task info with timeout
+     * @param timeout
+     * @param unit
+     * @return
+     * @throws TaskPriorityQueueException
+     * @throws InterruptedException
+     */
+    @Override
+    public TaskInstance poll(long timeout, TimeUnit unit) throws TaskPriorityQueueException, InterruptedException {
+        if (queue.isEmpty()) {
+            unit.sleep(timeout);
+        }
+        return queue.poll();
```

Comment:
why not use queue.poll(timeout, unit) directly?

Reply:
Because the `PriorityQueue` is used only the poll() method, there is no method with timeouts, while another `PriorityBlockingQueue` implements this method with timeouts.